test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/DerivedTypes.cs

```diff
@@ -0,0 +1,360 @@
+            {
+                return "null";
+            }
+
+            var encoder = UrlEncoder.Default;
+            var sb = new StringBuilder();
+            foreach(var item in data.Items)
+            {
+                sb.Append(encoder.UrlEncode(item.Key) + " " + encoder.UrlEncode(item.Value) + " ");
+            }
+
+            return sb.ToString();
+        }
+
+        AuthenticationProperties ISecureDataFormat<AuthenticationProperties>.Unprotect(string protectedText)
+        {
+            var propeties = new AuthenticationProperties();
+            if (protectedText != "null")
+            {
+                string[] items = protectedText.Split(' ');
+                for (int i = 0; i < items.Length - 1; i+=2)
+                {
+                    propeties.Items.Add(items[i], items[i + 1]);
+                }
+            }
+
+            return propeties;
+        }
+    }
+
+    /// <summary>
+    /// This formatter always throws. message and exception type can be set.
+    /// Defaults:
+    /// message: "AuthenticationPropertiesFormaterThrows"
+    /// type:  InvalidOperationException
+    /// </summary>
+    public class AuthenticationPropertiesFormaterThrows : ISecureDataFormat<AuthenticationProperties>
+    {
+        AuthenticationPropertiesFormaterThrows(string message = "AuthenticationPropertiesFormaterThrows", Type exceptionType = null)
+        {
+            Message = message;
+            ExceptionType = exceptionType ?? typeof(InvalidOperationException);
+        }
+
+        public Type ExceptionType { get; set; }
+
+        public string Message { get; set; }
+
+        public string Protect(AuthenticationProperties data)
+        {
+            throw (Exception)Activator.CreateInstance(ExceptionType, Message);
+        }
+
+        AuthenticationProperties ISecureDataFormat<AuthenticationProperties>.Unprotect(string protectedText)
+        {
+            throw (Exception)Activator.CreateInstance(ExceptionType, Message);
+        }
+    }
+
+    /// <summary>
+    /// This formatter returns values passed in the constructor
+    /// Defaults:
+    /// Protect: null
+    /// UnProtect: null
+    /// </summary>
+    public class AuthenticationPropertiesFormaterSetReturn : ISecureDataFormat<AuthenticationProperties>
+    {
+        string _protect;
+        AuthenticationProperties _unprotect;
+
+        AuthenticationPropertiesFormaterSetReturn(string protect = null, AuthenticationProperties unprotect = null)
+        {
+            _protect = protect;
+            _unprotect = unprotect;
+        }
+
+        public string Protect(AuthenticationProperties data)
+        {
+            return _protect;
+        }
+
+        AuthenticationProperties ISecureDataFormat<AuthenticationProperties>.Unprotect(string protectedText)
+        {
+            return _unprotect;
+        }
+    }
+
+    public class CustomConfigureOptions : ConfigureOptions<OpenIdConnectAuthenticationOptions>
```

Comment:
This doesn't seem to do anything, why preemptively introduce this before its needed?